docs/products/access/api-reference/custom-registration/custom-registration-v2.md

```diff
@@ -0,0 +1,166 @@
+
+JSON body parameters:
+
+| Param                   | Required                                | Description                                                                                                                                                  |
+|-------------------------|-----------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `client_assertion_type` | yes                                     | The type of the client assertion that is part of the request, currently the only supported value is `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` |
+| `client_assertion`      | yes                                     | Client assertion in the form of a JWT signed by the client's private key                                                                                     |
+| `transaction_id`        | yes (`TWO_STEP`) otherwise optional     | Generated in **Init step**. For `TWO_STEP`, ensures same transaction                                                                                         |
+| `data`                  | no                                      | Raw registration request data which will be provided to the Extension Engine                                                                                 |
+| `scope`                 | no                                      | An array of scopes. If none are specified the default scopes are granted.                                                                                    |
+
+Example request:
+
+```http
+POST /oauth/v2/custom-registration/example-custom-registration-idp/complete HTTP/1.1
+Host: onegini.example.com
+Content-Type: application/json
+
+{
+   "client_assertion_type" : "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+   "client_assertion" : "eyJhbGciOiJSUzI1NiIsImtpZCI6IjIyIn0.
+     eyJpc3Mi[...omitted for brevity...].
+     cC4hiUPo[...omitted for brevity...]",
+  "transaction_id": "123123",
+  "data": "{\"custom_json_key\":\"custom_ json data\"}",      //optional, e.g. challenge code response
+  "scope": ["read", "write"],
+}
+```
+
+Example successful response
+
+```http
+HTTP/1.1 200 Ok
+Content-Type: application/json;charset=UTF-8
+Cache-Control: no-store
+Pragma: no-cache
+
+{
+    "status": 2000,
+    "oauth_token": {
+        "token_type": "bearer",
+        "access_token": "8A5AB83A3C6B7AAC41471C1205167A35E0F9281ED277EE2FDE6E8DE30972936D",
+        "refresh_token": "8CAE26B2B8E8EC18B4D432886448C7F99B558063C517BA41F30966B37C104983",
+        "id_token": "eyJraWQiOiI1Nzk1[...omitted for brevity...]r9KM8c5y-Utpw",
+        "expires_in": 3600
+    },
+    "data": "{\"custom_json_key\":\"custom json data\"}", // optional
+}
+```
+
+In the event of an error in the Access Service, one of the following error codes will be returned:
+
+## Server error codes
```

Comment:
Is it correct that v2 has way more errors then v1?

Reply:
@andrei-one can you check this please?

Reply:
They have exactly the same amount of validations and errors. I updated v1 and checked the messages